--- facedetect.c	2012-05-06 13:30:46.820631295 +0200
+++ facedetect.c.oden	2012-05-06 13:30:59.962717408 +0200
@@ -28,7 +28,7 @@
  *
  * Every user visible function must have an entry in facedetect_functions[].
  */
-static function_entry facedetect_functions[] = {
+static zend_function_entry facedetect_functions[] = {
     PHP_FE(face_detect, NULL)
     PHP_FE(face_count, NULL)
     {NULL, NULL, NULL}
